eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/inmemory/InMemoryEventStorageEngine.java

```diff
@@ -153,7 +153,8 @@ public TrackingToken createTokenAt(Instant dateTime) {
                      .map(TrackedEventMessage::trackingToken)
                      .map(tt -> (GlobalSequenceTrackingToken) tt)
                      .map(tt -> new GlobalSequenceTrackingToken(tt.getGlobalIndex() - 1))
-                     .orElse(null);
+                     .map(tt -> (TrackingToken) tt)
+                     .orElse(createHeadToken());
```

Comment:
This construct causes "createHeadToken" to be called upon each call of "createTokenAt". That may be an expensive operation. Please use `orElseGet(this::createHeadToken)` instead.

Reply:
Darn, good point. Resolved.